src/main/java/net/sf/log4jdbc/Utilities.java

```diff
@@ -59,4 +62,32 @@ public static String rtrim(String s)
 		}
 		return s.substring(0, i + 1);
 	}
+
+	/**
+	 * Gets a config value from system properties and if not found, falls back to env properties.
+	 *
+	 * @param key key to get a config value for
+	 * @return config value, if found
+	 */
+	public static Optional<String> getConfigProp(String key) {
+		return Optional.ofNullable(System.getProperty(key)).or(() -> Optional.ofNullable(System.getenv().get(key)));
+	}
+
+	/**
+	 * Gets a config value as {@link #getConfigProp(String)} and converts it to the specified type using
+	 * the parser provided.
+	 *
+	 * @param key key to get a config value for
+	 * @param parser a function that converts config value of type string to the specified type
+	 * @return config value, if found, converted to the specified type
+	 * @param <T> a type to convert config value to
+	 */
+	public static <T> Optional<T> getConfigProp(String key, Function<String, T> parser) {
+		try {
+			// FIXME: parser reports errors by throwing exceptions which is not good in `map` method.

```

Comment:
FIXME?

Reply:
Yes, cutting corners because otherwise no lambdas + custom type for parser. Logging instead of throwing expected (checked) exceptions definitely solves the issue.

P.S. Even Scala [introducing checked exceptions](https://dotty.epfl.ch/docs/reference/experimental/canthrow.html), which proves that history goes in circles.

Reply:
Problem in this case for our given situation is that it will abort startup and spread havoc which is not ideal. Assuming the safest flag value is a better in our situation. Hence suggesting not throw exceptions.

Scala's story is a fun topic for us when we meet next and as you know the Scala one has more formal spec and intended for effect tracking/capturing.